### PR TITLE
mon/OSDMonitor:add "get CLI" for ratio (full)

### PR DESCRIPTION
**Pull request by yonghengdexin735**
add CLI:
ceph osd get-full-ratio
ceph osd get-backfillfull-ratio
ceph osd get-nearfull-ratio

Signed-off-by: yonghengdexin735 <zhang.zezhu@zte.com.cn>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -8312,6 +8312,21 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
...
+         << "luminous' before using the new interface";
+      err = -EPERM;
+      goto reply;
+    }
+    if (prefix == "osd get-full-ratio")
+      ss << "full_ratio is " << osdmap.get_full_ratio();
+    else if (prefix == "osd get-backfillfull-ratio")
+      ss << "backfillfull_ratio is " << osdmap.get_backfillfull_ratio();
+    else if (prefix == "osd get-nearfull-ratio")
+      ss << "nearfull_ratio is " << osdmap.get_nearfull_ratio()
```
Comment (gmayyyha) [inline src/mon/OSDMonitor.cc]: L8329 missing `;`

File: src/mon/MonCommands.h [hunk 2]
```diff
@@ -701,6 +701,18 @@ COMMAND("osd set-nearfull-ratio " \
 	"name=ratio,type=CephFloat,range=0.0|1.0", \
 	"set usage ratio at which OSDs are marked near-full",
 	"osd", "rw", "cli,rest")
+COMMAND("osd get-full-ratio " \
+	"name=ratio,type=CephFloat,range=0.0|1.0", \
+	"get usage ratio at which OSDs are marked full",
+	"osd", "rw", "cli,rest")
```
Comment (jecluis) [inline src/mon/MonCommands.h]: I don't like these commands, as they pollute the command namespace. I would prefer these to be under a specific category:

```
osd get full-ratio
osd get backfill-full-ratio
osd get near-full-ratio
```
Also, all these three commands are read-only, not read-write.

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -8312,6 +8312,21 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     wait_for_finished_proposal(op, new Monitor::C_Command(mon, op, 0, rs,
 					      get_last_committed() + 1));
     return true;
+  } else if (prefix == "osd get-full-ratio" ||
```
Comment (jecluis) [inline src/mon/OSDMonitor.cc]: All these commands are read-only. Please move them to `preprocess_command()`.

**Comment by jecluis** [comment]
On top of the required changes, this PR needs to be accompanied by tests and documentation for the new commands.

File: src/mon/OSDMonitor.cc (hunk 3 above)
Comment (yonghengdexin735) [reply src/mon/OSDMonitor.cc]: @jecluis done !

**Comment by tchaikov** [comment]
@yonghengdexin735 you are adding user-facing features. so as @jecluis pointed out, you need to add tests for it, and also document this features.

File: qa/workunits/cephtool/test.sh [hunk 4]
```diff
@@ -1820,10 +1820,13 @@ function test_mon_pg()
   ceph pg scrub 1.0
 
   ceph osd set-full-ratio .962
+  ceph osd get full-ratio
```
Comment (jecluis) [inline qa/workunits/cephtool/test.sh]: This is not validating the output, nor whether the command even succeeds. The same goes for all other `get` commands being tested.

**Comment by tchaikov** [comment]
@yonghengdexin7 ping?

**Pull request closed without merging**